src/WebJobs.Script/Config/FunctionsHostingEnvironmentConfigSource.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Configuration.Json;
+
+namespace Microsoft.Azure.WebJobs.Script.Config
+{
+    /// <summary>
+    /// Represents a JSON file as an <see cref="IConfigurationSource"/>.
+    /// </summary>
+    public class FunctionsHostingEnvironmentConfigSource : JsonConfigurationSource
+    {
+        private readonly string _path;
+
+        public FunctionsHostingEnvironmentConfigSource(IEnvironment environment)
+        {
+            _path = environment.GetEnvironmentVariable(EnvironmentSettingNames.FunctionsHostingEnvironmentConfigFilePath);
```

Comment:
What to do if _path is null?

Reply:
The path here can be null. The file has been marked optional. In case path is null, IConfigurationProvider will not read the file. I have added a test case for that scenario.